cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -500,6 +473,13 @@ public OutputStream getOutput() throws IOException {
   }
 
   /**
+   * Get the instance of the {@link HBaseDDLExecutor}.
+   * @param hConf the hadoop configurations
+   * @return the instance of DDL executor
+   */
+  public abstract HBaseDDLExecutor getHBaseDDLExecutor(Configuration hConf);
```

Comment:
when you make the DDLExecutor pluggable, creation of the DDLExecutor can't be done by the children of this class. In fact, it seems more natural for the default DDLExecutor to use HBaseTableUtil to perform its operations, though I'm not entirely sure how we let it set up the guice injector. Do you know how we instantiate the classes defined in the security spi?